### ISSUE TITLE
Word 365: NVDA is silent when pressing Escape in a Word ribbon

### ISSUE THREAD

**Issue by bhavyashah** [nvaccess/nvda#13073]
### Steps to reproduce:
1 Open a Word document.
2 Press Alt+H.
3 Press Esc.

### Actual behavior:
NVDA is silent.

### Expected behavior:
I cannot be certain, but my guess is that the first Escape does havve some effect which NVDA is not but should be reporting.

### Additional Notes
Pressing Alt followed by Escape gets you back to the editing area and NVDA reports it just fine. However, on pressing Alt+H or Alt+N or such, two Escapes are necessary to get out of the ribbon and back in the editing area. That is Word behaviour and need not be modified by NVDA. But the effect of the first Escape is important for the user to know so that they don't incorrectly believe that they are back in the editing area and perhaps NVDA failed to report that. Also, this issue occurs with Word UIA enabled as well as disabled.

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA 2021.3 Beta 2
#### Windows version:
Windows 10
#### Name and version of other software in use when reproducing the issue:
Word for Microsoft 365 64 bit
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by Qchristensen** [comment]
When you press ALT, the focus moves to the ribbon header, ready to arrow to Home, Insert, Draw, Design, or whichever ribbon you want.  At this point, the keys to access the quick access toolbar, or each ribbon are displayed. 
 Pressing ESCAPE at that point moves the focus back to the document.

When you press ALT+H, the focus jumps directly to the Home ribbon (or ALT+N goes to the insert ribbon etc).  At this point the shortcut keys for each item in the Home (or insert, etc) ribbon are highlighted visually.  Pressing ESCAPE at this point moves the focus back to the ribbon header - just as if you'd pressed ALT to begin with.

Interestingly if the ribbon is set to auto hide (press control+1 to toggle), the ribbon is reported when alt is pressed, it is not reported when alt+h is pressed, but if you press alt+h, then escape to return to the ribbon headers, THEN NVDA reports that the focus is in the ribbon.

Using NVDA 2021.3 beta 2 and Office 365 (64-bit) Version: 16.0.14527.20144, all of these happens whether or not UIA is enabled.  Here is the log fragment from pressing alt+h then escape twice:

IO - inputCore.InputManager.executeGesture (09:58:08.081) - winInputHook (13548):
Input: kb(desktop):alt+h
IO - speech.speech.speak (09:58:08.469) - MainThread (2800):
Speaking [LangChangeCommand ('en_GB'), 'Ribbon Tabs', 'tab control', 'expanded', CancellableSpeech (still valid)]
IO - speech.speech.speak (09:58:08.527) - MainThread (2800):
Speaking [LangChangeCommand ('en_GB'), 'Home', 'tab', 'selected', 'Alt, H', '2 of 13', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (09:58:10.108) - winInputHook (13548):
Input: kb(desktop):escape
IO - inputCore.InputManager.executeGesture (09:58:14.081) - winInputHook (13548):
Input: kb(desktop):escape
DEBUG - NVDAObjects.NVDAObject._get_placeholder (09:58:14.197) - MainThread (2800):
Potential unimplemented child class: <NVDAObjects.Dynamic_WinwordWordDocumentWordDocumentEditableTextWithAutoSelectDetectionUIAWindowNVDAObject object at 0x06ABDA30>
IO - speech.speech.speak (09:58:14.213) - MainThread (2800):
Speaking [LangChangeCommand ('en_GB'), 'document', CancellableSpeech (still valid), 'page 1', 'blank']

**Comment by bhavyashah** [comment]
This issue is reproducible on Word 365 (latest updates installed) and with NVDA 2023.2.

**Comment by Adriani90** [comment]
Still reproducible with NVDA 2025.3 and MS Word 365 version 2509.
This needs a notification introduced by Microsoft to tell NVDA what happens.
cc: @msftedad can you please report this issue to the development team?

**Comment by msftedad** [comment]
@Adriani90, We have forwarded this issue to development team. We will provide you with the tracking id once we receive update from them. Thanks

**Comment by msftedad** [comment]
Issue shared with dependent team, for reference, team have created a ticket "[Accessibility 249492](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2FCSS-IRT%2FCSS%2520IRT%2F_workitems%2Fedit%2F249492&data=05%7C02%7Cv-ymounika%40microsoft.com%7C9430f8940bdf4da878fb08de126d6ac7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638968458371140057%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=hLv7lreT%2BpseXHYUJVgHSWly5t6L3BgrxCFiTn4QUPk%3D&reserved=0)" tracking purposes, you won’t have access to this link, but this can be used as a reference number for this request.

**Comment by msftedad** [comment]
We have verified and confirmed with the engineering team that this behavior is by design. Screen readers announce focus changes when navigating the ribbon, which is the essential requirement. You will only hear an announcement if there is a change in focus. Announcing the appearance or disappearance of key tips is not necessary, as it does not impact the user’s ability to operate the interface.

Please let us know if there are further questions. Thanks
